Apps/Src/ReadCarCAN.c

```diff
@@ -212,28 +152,38 @@ void Task_ReadCarCAN(void *p_arg)
 
                     // Create an exception and assert the error
                     // kill contactors and enter a nonrecoverable fault
-                    exception_t tripBPSError = {.prio=1, .message="BPS has been tripped", .callback=callback_disableContactors};
+                    exception_t tripBPSError = {.prio=1, .message="BPS has been tripped", .callback=NULL};
                     assertExceptionError(tripBPSError);
                 }
             }
-            case CHARGE_ENABLE: { 
+            case CHARGE_ENABLE:{
 
                 // Restart CAN Watchdog timer
                 OSTmrStart(&canWatchTimer, &err);
                 assertOSError(OS_READ_CAN_LOC, err);
 
+                if(dataBuf.data[0] == 0){
+
+                    // Disable regen and update saturation in callback function
+                    exception_t disableCharging = {.prio=2, .message="Charging needs to be disabled", .callback=callback_chargingDisable};
+                    assertExceptionError(disableCharging);
 
-                if(dataBuf.data[0] == 0){ // If the buffer doesn't contain 1 for enable, turn off chargeEnable and turn array off
-                    chargingDisable();
-                    updateSaturation(-1); // Update saturation and buffer
+                } else {
 
-                } else { //We got a message of enable with a nonzero value
-                    updateSaturation(1);
+                    // Enable regen and update saturation
+                    updateSaturation(1);                               
+                    
+                    if(Minion_Read_Pin(IGN_1, &Merr) == ON                                          // Ignition is ON
+                        && chargeMsgSaturation >= SATURATION_THRESHOLD                              // Saturation Threshold has be met
+                        && (Contactors_Get(ARRAY_CONTACTOR)==OFF)                                   // Array Contactor is OFF
+                        && OSTmrStateGet(&prechargeDlyTimer, &err) != OS_TMR_STATE_STOPPED){        // Precharge Delay is not runnning and is not executing one-shot 
```

Comment:
Thank you :) I was debating between using either (1) STOPPED or (2) RUNNING and COMPLETED. They should have opposite functionality. I ultimately decided on STOPPED because it would be one statement but I forgot to change the negation.